### PR TITLE
CI: migrate pipeline into GH Actions

### PR DESCRIPTION
**Pull request by brushknight**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows (keep old codes)
- [x] Squash commits because Joan Porta did not accept CLA
- [x] Use org Slack secrets

### CODE REVIEW

**Review by paologallinaharbur — APPROVED** [review]
Again everything looks good, I have some doubts on the GUID

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 1]
```diff
@@ -49,47 +49,50 @@
...
         </DirectoryRef>
         <DirectoryRef Id="CustomPluginsFolder">
-            <Component Id="CMP_V1_CUSTOM_PLUGINS" Guid="0EE4AAF4-9923-499C-A1D0-FF55B205A76C" KeyPath="yes">
+            <Component Id="CMP_V1_CUSTOM_PLUGINS" Guid="1f8a618a-a5c9-4828-abfe-627f4faaf830" KeyPath="yes">
                 <CreateFolder/>
             </Component>
         </DirectoryRef>
         <DirectoryRef Id="CustomPluginsBinFolder">
-            <Component Id="CMP_V1_CUSTOM_PLUGINS_BIN" Guid="e4509a7a-409e-4a38-8705-37de9cf0d72d" KeyPath="yes">
+            <Component Id="CMP_V1_CUSTOM_PLUGINS_BIN" Guid="a9351e80-0bb7-4d32-aca0-6529e3eea905" KeyPath="yes">
```
Comment (roobre) [inline build/package/windows/nri-amd64-installer/Product.wxs:62]: Suspicious GUID change

Comment (roobre) [reply build/package/windows/nri-amd64-installer/Product.wxs:62]: These IDs were the same across all repos

**Comment by roobre** [comment]
GUID changes reverted, please double-check

**Review by paologallinaharbur — APPROVED** [review]
lgtm

**Pull request merged**